### PR TITLE
Split imagereferencemutators to origin and non-origin parts

### PR DESCRIPTION
**Pull request by soltysh**

### CODE REVIEW

File: pkg/image/apiserver/admission/imagepolicy/imagereferencemutators/interfaces.go [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+	kapiv1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	kapi "k8s.io/kubernetes/pkg/apis/core"
+)
+
+type GetPodSpecFunc func(obj runtime.Object) (*kapi.PodSpec, *field.Path, error)
+
+var GetPodSpecFn GetPodSpecFunc = GetPodSpec
```
Comment (deads2k) [inline pkg/image/apiserver/admission/imagepolicy/imagereferencemutators/interfaces.go]: It think refactor this into an admission plugin initializer with 

```go
type ImageReferenceMutatorFactor interface{
    GetImageReferenceMutator
    GetAnnotationAccessor
}
```

File: pkg/image/apiserver/admission/imagepolicy/imagereferencemutators/pods.go [hunk 2]
```diff
@@ -8,31 +8,15 @@ import (
...
-
-type PodSpecReferenceMutator interface {
-	GetContainerByIndex(init bool, i int) (ContainerMutator, bool)
-	GetContainerByName(name string) (ContainerMutator, bool)
-	Path() *field.Path
-}
-
 // GetPodSpecReferenceMutator returns a mutator for the provided object, or an error if no
 // such mutator is defined.
 func GetPodSpecReferenceMutator(obj runtime.Object) (PodSpecReferenceMutator, error) {
```
Comment (deads2k) [inline pkg/image/apiserver/admission/imagepolicy/imagereferencemutators/pods.go]: I expected this file to land somewhere in `pkg/cmd/openshift-kube-apiserver/admission`, but it looks like the right idea.

File: pkg/cmd/openshift-kube-apiserver/admission/imagepolicy/imagepolicy.go [hunk 3]
```diff
@@ -62,28 +62,31 @@ func Register(plugins *admission.Plugins) {
 				return nil, errs.ToAggregate()
 			}
 			klog.V(5).Infof("%s admission controller loaded with config: %#v", imagepolicy.PluginName, config)
-			return newImagePolicyPlugin(config)
+			return NewImagePolicyPlugin(config)
 		})
 }
 
-type imagePolicyPlugin struct {
+type ImagePolicyPlugin struct {
```
Comment (deads2k) [inline pkg/cmd/openshift-kube-apiserver/admission/imagepolicy/imagepolicy.go]: I'm surprised by this, but I'm still reading.

Comment (soltysh) [reply pkg/cmd/openshift-kube-apiserver/admission/imagepolicy/imagepolicy.go]: The split required me to be able to inject it inside values for testing. That was the simplest, iow. my laziness :wink:

File: pkg/cmd/server/admission/init.go [hunk 4]
```diff
@@ -25,6 +26,7 @@ type PluginInitializer struct {
 	SecurityInformers              securityv1informer.SecurityContextConstraintsInformer
 	UserInformers                  userinformer.SharedInformerFactory
 	RoleBindingRestrictionInformer authorizationinformer.RoleBindingRestrictionInformer
+	ImageMutators                  imagereferencemutators.ImageMutators
```
Comment (deads2k) [inline pkg/cmd/server/admission/init.go]: note to self, break these things apart.  this is crazy.

**Comment by deads2k** [comment]
fixed up in https://github.com/openshift/origin/pull/23179 so it can merge while @soltysh is out.

**Pull request closed without merging**